datalad/distribution/tests/test_install.py

```diff
@@ -501,3 +510,47 @@ def test_reckless(path, top_path):
     ds = install(path=top_path, source=path, reckless=True)
     eq_(ds.config.get('annex.hardlink', None), 'true')
     eq_(ds.repo.repo_info()['untrusted repositories'][0]['here'], True)
+
+
+@with_tree(tree={'top_file.txt': 'some',
+                 'sub 1': {'sub1file.txt': 'something else',
+                           'subsub': {'subsubfile.txt': 'completely different',
+                                      }
+                           },
+                 'sub 2': {'sub2file.txt': 'meaningless',
+                           }
+                 })
+@with_tempfile(mkdir=True)
+def test_install_recursive_repeat(src, path):
+    subsub_src = Dataset(opj(src, 'sub 1', 'subsub')).create(force=True)
+    sub1_src = Dataset(opj(src, 'sub 1')).create(force=True)
+    sub2_src = Dataset(opj(src, 'sub 2')).create(force=True)
+    top_src = Dataset(src).create(force=True)
+    top_src.save(auto_add_changes=True, recursive=True)
+
+    # install top level:
+    top_ds = install(path=path, source=src)
```

Comment:
we could also test cmdline API by simply calling `datalad.cmdline.main:main(['datalad', 'install', '--source', ...]`